Bug Postgres bool

v1/rdbms/gatus/endpoint_results?filter=success==false

All other filter types are working string int - only bool is not

From the little investigation i did, looks it's a cast issue:
- false is interpreted as a java type string and not a boolean (log error: operator does not exist: boolean = character varying)
- postgres is sensitive about types
- unit test seems don't cover bool?

> [!NOTE]
> Schema included at the end


```log

2026-04-14T10:48:18.514Z  INFO 1 --- [mcat-handler-73] .h.d.i.DatabaseContextRequestInterceptor : Pre handle - /v1/rdbms/gatus/endpoint_results
2026-04-14T10:48:18.515Z  INFO 1 --- [mcat-handler-73] .h.d.i.DatabaseContextRequestInterceptor : pathVariables - {dbId=gatus, tableName=endpoint_results}
2026-04-14T10:48:18.528Z ERROR 1 --- [mcat-handler-73] c.h.d.jdbc.core.service.JdbcReadService  : Error in read op :

org.springframework.jdbc.BadSqlGrammarException: PreparedStatementCallback; bad SQL grammar [
SELECT
    endp_619aa4d36ac7."endpoint_result_id"
	,endp_619aa4d36ac7."endpoint_id"
	,endp_619aa4d36ac7."success"
	,endp_619aa4d36ac7."errors"
	,endp_619aa4d36ac7."connected"
	,endp_619aa4d36ac7."status"
	,endp_619aa4d36ac7."dns_rcode"
	,endp_619aa4d36ac7."certificate_expiration"
	,endp_619aa4d36ac7."domain_expiration"
	,endp_619aa4d36ac7."hostname"
	,endp_619aa4d36ac7."ip"
	,endp_619aa4d36ac7."duration"
	,endp_619aa4d36ac7."timestamp"
	,endp_619aa4d36ac7."suite_result_id"
FROM
    public.endpoint_results endp_619aa4d36ac7


    WHERE endp_619aa4d36ac7.success = ?



    LIMIT 100


]
	at org.springframework.jdbc.support.SQLStateSQLExceptionTranslator.doTranslate(SQLStateSQLExceptionTranslator.java:112) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:107) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:116) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.core.JdbcTemplate.translateException(JdbcTemplate.java:1556) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:677) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:723) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:748) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.core.JdbcTemplate.query(JdbcTemplate.java:804) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate.query(NamedParameterJdbcTemplate.java:218) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at com.homihq.db2rest.jdbc.JdbcOperationService.read(JdbcOperationService.java:42) ~[rdbms-support-1.6.8.jar!/:1.6.8]
	at com.homihq.db2rest.jdbc.core.service.JdbcReadService.findAll(JdbcReadService.java:37) ~[rdbms-support-1.6.8.jar!/:1.6.8]
	at com.homihq.db2rest.rest.read.ReadController.findAll(ReadController.java:55) ~[!/:1.6.8]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:258) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:191) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:118) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:991) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:896) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:564) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.2.10.jar!/:6.2.10]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:195) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-embed-websocket-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.springframework.web.filter.ServerHttpObservationFilter.doFilterInternal(ServerHttpObservationFilter.java:110) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.springframework.web.filter.ForwardedHeaderFilter.doFilterInternal(ForwardedHeaderFilter.java:173) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.2.10.jar!/:6.2.10]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:483) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:116) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:398) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:903) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1769) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.44.jar!/:na]
	at java.base/java.lang.VirtualThread.run(Unknown Source) ~[na:na]
Caused by: org.postgresql.util.PSQLException: ERROR: operator does not exist: boolean = character varying
  Hint: No operator matches the given name and argument types. You might need to add explicit type casts.
  Position: 569
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2734) ~[postgresql-42.7.7.jar!/:42.7.7]
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2421) ~[postgresql-42.7.7.jar!/:42.7.7]
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372) ~[postgresql-42.7.7.jar!/:42.7.7]
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:518) ~[postgresql-42.7.7.jar!/:42.7.7]
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:435) ~[postgresql-42.7.7.jar!/:42.7.7]
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:196) ~[postgresql-42.7.7.jar!/:42.7.7]
	at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:139) ~[postgresql-42.7.7.jar!/:42.7.7]
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52) ~[HikariCP-6.3.2.jar!/:na]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java) ~[HikariCP-6.3.2.jar!/:na]
	at org.springframework.jdbc.core.JdbcTemplate$1.doInPreparedStatement(JdbcTemplate.java:732) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:658) ~[spring-jdbc-6.2.10.jar!/:6.2.10]
```

v1/rdbms/gatus/$schemas?columns=true

```json
 {
    "schema": "public",
    "name": "endpoint_results",
    "type": "TABLE",
    "columns": [
      {
        "name": "endpoint_result_id",
        "pk": true,
        "dataType": "bigserial"
      },
      {
        "name": "endpoint_id",
        "pk": false,
        "dataType": "int8"
      },
      {
        "name": "success",
        "pk": false,
        "dataType": "bool"
      },
      {
        "name": "errors",
        "pk": false,
        "dataType": "text"
      },
      {
        "name": "connected",
        "pk": false,
        "dataType": "bool"
      },
      {
        "name": "status",
        "pk": false,
        "dataType": "int8"
      },
      {
        "name": "dns_rcode",
        "pk": false,
        "dataType": "text"
      },
      {
        "name": "certificate_expiration",
        "pk": false,
        "dataType": "int8"
      },
      {
        "name": "domain_expiration",
        "pk": false,
        "dataType": "int8"
      },
      {
        "name": "hostname",
        "pk": false,
        "dataType": "text"
      },
      {
        "name": "ip",
        "pk": false,
        "dataType": "text"
      },
      {
        "name": "duration",
        "pk": false,
        "dataType": "int8"
      },
      {
        "name": "timestamp",
        "pk": false,
        "dataType": "timestamp"
      },
      {
        "name": "suite_result_id",
        "pk": false,
        "dataType": "int8"
      }
    ]
  },
```

--- a/db2rest-api/api-rest/src/test/java/com/homihq/db2rest/rest/pg/PgReadControllerTest.java
+++ b/db2rest-api/api-rest/src/test/java/com/homihq/db2rest/rest/pg/PgReadControllerTest.java
@@ -108,7 +108,7 @@ class PgReadControllerTest extends PostgreSQLBaseIntegrationTest {
     @Test
     @DisplayName("Test PostgreSQL boolean type casting in filter")
     void testBooleanFilterCasting() throws Exception {
-        mockMvc.perform(get("/v1/rdbms/pgsqldb/users?filter=isActive==true")
+        mockMvc.perform(get(VERSION + "/pgsqldb/users?filter=isActive==true")
                         .header("Accept-Profile", "public"))
                 .andExpect(status().isOk())
                 .andExpect(jsonPath("$.[*]").isArray())

--- a/db2rest-api/api-rest/src/test/java/com/homihq/db2rest/rest/pg/PgReadControllerTest.java
+++ b/db2rest-api/api-rest/src/test/java/com/homihq/db2rest/rest/pg/PgReadControllerTest.java
@@ -13,6 +13,7 @@ import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestClassOrder;
 import org.springframework.http.MediaType;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
 
 import java.util.Map;
 
@@ -102,5 +103,17 @@ class PgReadControllerTest extends PostgreSQLBaseIntegrationTest {
                 .andExpect(status().isOk())
                 //.andDo(print())
                 .andDo(document("pg-get-one-film"));
+    }
+
+    @Test
+    @DisplayName("Test PostgreSQL boolean type casting in filter")
+    void testBooleanFilterCasting() throws Exception {
+        mockMvc.perform(get("/v1/rdbms/pgsqldb/users?filter=isActive==true")
+                        .header("Accept-Profile", "public"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.[*]").isArray())
+                // Verify the first item actually parsed the boolean correctly
+                .andExpect(jsonPath("$.[0].isactive").value(true))
+                .andDo(print());
     }
 }

--- a/db2rest-core/rdbms-common/src/main/java/com/db2rest/jdbc/dialect/Dialect.java
+++ b/db2rest-core/rdbms-common/src/main/java/com/db2rest/jdbc/dialect/Dialect.java
@@ -71,16 +71,9 @@ public abstract class Dialect {
         if (String.class == type) {
             //return "'" + value + "'";
             return value;
-        }
-         else if (Boolean.class == type || boolean.class == type) {
+        } else if (Boolean.class == type || boolean.class == type) {
             return Boolean.valueOf(value);
-        }
-//        else if (Boolean.class == type || boolean.class == type) {
-//            Boolean aBoolean = Boolean.valueOf(value);
-//            return aBoolean ? "1" : "0";
-//        }
-
-        else if (Integer.class == type || int.class == type) {
+        } else if (Integer.class == type || int.class == type) {
             return Integer.valueOf(value);
         } else if (Long.class == type || long.class == type) {
             return Long.valueOf(value);

--- a/db2rest-core/rdbms-common/src/main/java/com/db2rest/jdbc/dialect/Dialect.java
+++ b/db2rest-core/rdbms-common/src/main/java/com/db2rest/jdbc/dialect/Dialect.java
@@ -71,10 +71,16 @@ public abstract class Dialect {
         if (String.class == type) {
             //return "'" + value + "'";
             return value;
-        } else if (Boolean.class == type || boolean.class == type) {
-            Boolean aBoolean = Boolean.valueOf(value);
-            return aBoolean ? "1" : "0";
-        } else if (Integer.class == type || int.class == type) {
+        }
+         else if (Boolean.class == type || boolean.class == type) {
+            return Boolean.valueOf(value);
+        }
+//        else if (Boolean.class == type || boolean.class == type) {
+//            Boolean aBoolean = Boolean.valueOf(value);
+//            return aBoolean ? "1" : "0";
+//        }
+
+        else if (Integer.class == type || int.class == type) {
             return Integer.valueOf(value);
         } else if (Long.class == type || long.class == type) {
             return Long.valueOf(value);

